pkg/cloudprovider/plugins/azure/azure_compute.go

```diff
@@ -363,31 +419,45 @@ func (computeCfg *computeServiceConfig) buildMapVpcPeers(results []armnetwork.Vi
 	return vpcPeers
 }
 
-// GetVpcInventory generates vpc object for the vpcs stored in snapshot(in cloud format) and return a map of vpc runtime objects.
-func (computeCfg *computeServiceConfig) GetVpcInventory() map[string]*runtimev1alpha1.Vpc {
+// GetCloudInventory fetches VM and VPC inventory from stored snapshot and converts from cloud format to internal format.
+func (computeCfg *computeServiceConfig) GetCloudInventory() *nephetypes.CloudInventory {
+	cloudInventory := nephetypes.CloudInventory{
+		VmMap:  map[types.NamespacedName]map[string]*runtimev1alpha1.VirtualMachine{},
+		VpcMap: map[string]*runtimev1alpha1.Vpc{},
+	}
+	cloudInventory.VpcMap = computeCfg.GetVpcObjects()
+	for ns := range computeCfg.selectorConfigMap {
+		cloudInventory.VmMap[ns] = computeCfg.getVirtualMachineObjects(&computeCfg.accountNamespacedName, &ns)
+	}
+
+	return &cloudInventory
+}
+
+// GetVpcObjects generates vpc object for the vpcs stored in snapshot(in cloud format) and return a map of vpc runtime objects.
+func (computeCfg *computeServiceConfig) GetVpcObjects() map[string]*runtimev1alpha1.Vpc {
+	vnetIDs := computeCfg.getManagedVnetIDs()
 	snapshot := computeCfg.resourcesCache.GetSnapshot()
 	if snapshot == nil {
-		azurePluginLogger().Info("Compute service cache snapshot nil",
-			"type", providerType, "account", computeCfg.account)
+		azurePluginLogger().Info("Cache snapshot nil",
+			"type", providerType, "account", computeCfg.accountNamespacedName)
 		return nil
 	}
 
-	vnetIDs := computeCfg.getManagedVnetIDs()
-
 	// Convert to kubernetes object and return a map indexed using VnetID.
 	vpcMap := map[string]*runtimev1alpha1.Vpc{}
-	for _, vpc := range snapshot.(*computeResourcesCacheSnapshot).vnets {
+	for _, vpc := range snapshot.(*cacheSnapshot).vnets {
 		if strings.EqualFold(*vpc.Location, computeCfg.credentials.region) {
 			managed := false
 			if _, ok := vnetIDs[strings.ToLower(*vpc.ID)]; ok {
 				managed = true
 			}
-			vpcObj := ComputeVpcToInternalVpcObject(&vpc, computeCfg.account.Namespace, computeCfg.account.Name,
-				strings.ToLower(computeCfg.credentials.region), managed)
+			vpcObj := ComputeVpcToInternalVpcObject(&vpc, computeCfg.accountNamespacedName.Namespace,
+				computeCfg.accountNamespacedName.Name, strings.ToLower(computeCfg.credentials.region), managed)
 			vpcMap[strings.ToLower(*vpc.ID)] = vpcObj
 		}
 	}
-	azurePluginLogger().V(1).Info("Cached vpcs", "account", computeCfg.account, "vpc objects", len(vpcMap))
+	azurePluginLogger().V(1).Info("Cached vpcs", "account", computeCfg.accountNamespacedName,
```

Comment:
Seems a redundant log message. Please check if this can be removed.

Reply:
log while retrieving vm/vpcs from snapshot has been removed when the call is made from poller.